platform/linux-generic/odp_schedule_basic.c

```diff
@@ -1339,8 +1339,10 @@ static inline int schedule_grp_prio(odp_queue_t *out_queue, odp_event_t out_ev[]
 				 * packet input queue even when it is empty. */
 				ring_u32_enq(ring, ring_mask, qi);
 
-				/* Continue scheduling from the next group/priority */
-				return 0;
+				/* Continue scheduling from the next spread */
+				i++;
+				spr++;
```

Comment:
I am not convinved that this is the right thing to do. Not necessarily because it is not but maybe because I have hard time thinking about the dynamics at play and seeing whether this changes it for the better or worse in various situations. I guess I am trying to say that this is hard for me to review.

It seems to me that with ordered pktio input this piece of code selects the next spread even when there are packets available in the pktio (but they just have to fo through the internal queue first). I wonder if changing the behaviour in that particular case is good. More generally, in cases when packet I/Os are not getting more traffic that can be processed, this change causes the spread to change often even if there might be regular non-pktio queues that have packets available. This seems to reduce the possible benefit of having the spreads in the first place. Or maybe I just got confused.